packages/solid-urql/src/createQuery.ts

```diff
@@ -0,0 +1,334 @@
+ * context’s {@link Client}.
+ *
+ * The returned result updates when the `Client` has new results
+ * for the query, and changes when your input `args` change.
+ *
+ * Additionally, if the `suspense` option is enabled on the `Client`,
+ * the `createQuery` hook will suspend instead of indicating that it’s
+ * waiting for a result via {@link CreateQueryState.fetching}.
+ *
+ * @example
+ * ```tsx
+ * import { gql, createQuery } from '@urql/solid';
+ *
+ * const TodosQuery = gql`
+ *   query { todos { id, title } }
+ * `;
+ *
+ * const Todos = () => {
+ *   const [result, reExecuteQuery] = createQuery({
+ *     query: TodosQuery,
+ *   });
+ *   // ...
+ * };
+ * ```
+ */
+export const createQuery = <
+  Data = any,
+  Variables extends AnyVariables = AnyVariables
+>(
+  args: CreateQueryArgs<Data, Variables>
+): CreateQueryResult => {
+  const client = useClient();
+  const getContext = asAccessor(args.context),
+    getPause = asAccessor(args.pause),
+    getRequestPolicy = asAccessor(args.requestPolicy),
+    getVariables = asAccessor(args.variables);
+
+  const resultSourceSubject = makeSubject<
+    OperationResultSource<OperationResult<Data, Variables>> | undefined
+  >();
+
+  // Combine suspense param coming from context and client with context being priority
+  const isSuspense = createMemo(() => {
+    const ctx = getContext();
+    if (ctx !== undefined && ctx.suspense !== undefined) {
+      return ctx.suspense;
+    }
+
+    return client.suspense;
+  });
+
+  const request = createRequest(args.query, getVariables() as any);
+  const context: Partial<OperationContext> = {
+    requestPolicy: getRequestPolicy(),
+    ...getContext(),
+  };
+  const operation = client.createRequestOperation('query', request, context);
+  const initialResult: CreateQueryState<Data, Variables> = {
+    operation: operation,
+    fetching: false,
+    data: undefined,
+    error: undefined,
+    extensions: undefined,
+    hasNext: false,
+    stale: false,
+  };
+
+  const [result, setResult] =
+    createStore<CreateQueryState<Data, Variables>>(initialResult);
+
+  // `createResource` is only way to trigger suspense, suspense handling works as follows:
+  // 1. Calling `dataResource` mounts resource to closes suspense. Thus depending on where it is called
+  // it could mount in same component or parent. For usage convince we call it by proxy when using `data`
+  // 2. After any changes we call `refetch` which results in creation of new promise and depending on `fetching`
+  // value we either discard promise without resolving or resolve it. Discard doesn't result in anything while
+  // resolve finishes suspense
+  const [dataResource, { refetch, mutate }] = createResource<
+    CreateQueryState<Data, Variables>
+  >(() => {
+    return new Promise(resolve => {
+      const got = unwrap(result);
+      if (got.fetching) {
+        return;
+      }
+
+      resolve(got);
+    });
+  });
+
+  const sub = pipe(
+    resultSourceSubject.source,
```

Comment:
I'm not sure whether using a subject here makes sense. It may make more sense to actually put the source in a store, similar to Vue again. This depends, but generally, I've found that using a subject here makes it actually a little harder to handle the control flow.

Reply:
I don't have preferences both solutions have some tradeoffs - source state  + effect + cleanup / subject + switchMap. You can take a look at source state implementation in https://github.com/urql-graphql/urql/pull/3327/commits/604c6a52bc2b4b93d5cfc784d3236ebcf6a5a13b